temba/channels/models.py

```diff
@@ -1424,11 +1427,12 @@ def send_junebug_message(cls, channel, msg, text):
         payload['content'] = text
 
         if is_ussd:
-            external_id, session_status = Msg.objects.values_list('response_to__external_id', 'session__status').get(pk=msg.id)
+            session = USSDSession.objects.get_session_with_status_only(msg.session_id)
+            external_id = Msg.objects.values_list('external_id', flat=True).filter(pk=msg.response_to_id).first()
             # NOTE: Only one of `to` or `reply_to` may be specified
             payload['reply_to'] = external_id
             payload['channel_data'] = {
-                'continue_session': session_status not in ChannelSession.DONE,
+                'continue_session': session and not session.should_end,
```

Comment:
@nullcode there is a very subtle bug here: `session` can be `None` which would set `continue_session` to be `None` where only `True` or `False` are valid values.